Observer/Hooks.php

```diff
@@ -45,6 +45,7 @@ public function mobbexCheckoutRequest($body, $orderId)
                 $fee         += $this->helper->getCommission($item);
                 $shipping     = $shipping ?: $this->helper->getVendorOrder($item)->getShippingInclTax();
                 $productIds[] = $product->getId();
+                $merchants    = $this->helper->getVendorEntity($item);
```

Comment:
Dentro de este loop se esta constantemente reasignando el valor de $merchants, teniendo en cuenta que el método "getVendorsEntity" retorna algo como: ['entity' => 'bFY765']
el valor merchants en cada iteración se reasigna con la entidad asigna al item de esa iteración.
Si lo que queres es que esa iteración se sume a un array de merchants tendria que haber un $merchants[] = $this->helper->getVendorEntity($item); con lo que indicarias que esa queres que sea la ultima posicion de merchants y merchants pasaria a ser segun lo que retorna tu metodo:

$merchants = [ ['entity' => 'HkJ65'], ['entity' => 'IJF609'] ]; 
por ejemplo.

Reply:
Esto pude corroborar en el flujo que se guarda correctamente. Hice unas modificaciones en el método para simplificarlo